crates/router/src/core/payments/flows/setup_mandate_flow.rs

```diff
@@ -303,6 +254,130 @@ impl types::SetupMandateRouterData {
             _ => Ok(self.clone()),
         }
     }
+
+    async fn update_mandate_flow(
+        self,
+        state: &AppState,
+        merchant_account: &domain::MerchantAccount,
+        mandate_id: String,
+        connector: &api::ConnectorData,
+        key_store: &domain::MerchantKeyStore,
+        call_connector_action: payments::CallConnectorAction,
+        supported_connectors_for_update_mandate: &settings::SupportedPaymentMethodsForMandate,
+        connector_request: Option<services::Request>,
+        maybe_customer: &Option<domain::Customer>,
+    ) -> RouterResult<Self> {
+        let payment_method_type = self.request.payment_method_type;
+
+        let payment_method = self
+            .request
+            .payment_method_data
+            .get_payment_method_for_payment_method_data();
+
+        if payment_method_type.map_or(false, |pmt| {
```

Comment:
Put the condition result in a variable and use the variable in the if statement.

Reply:
Also
```suggestion
        payment_method.zip(payment_method_type).map_or(false, |(pm, pmt)| { ... })
```
is a better way to do it.